test/com/amazon/ion/impl/IonReaderBinaryRawLargeStreamTest.java

```diff
@@ -70,4 +79,298 @@ public void testReadLargeScalarStream() throws Exception {
+            }
+            batchesRead++;
+        }
+        assertNull(reader.next());
+        reader.stepOut();
+        assertNull(reader.next());
+        assertEquals(totalNumberOfBatches, batchesRead);
+    }
+
+    @Test
+    public void skipLargeAnnotatedContainer() throws Exception {
+        final Timestamp timestamp = Timestamp.forDay(2000, 1, 1);
+        byte[] raw = testData(timestamp);
+        ByteArrayOutputStream dataBuilder = new ByteArrayOutputStream();
+        dataBuilder.write(0xEE); // Annotation wrapper with length subfield.
+        IonBinary.writeVarUInt(dataBuilder, 3 + IonBinary.lenVarUInt(raw.length) + raw.length);
+        dataBuilder.write(0x81); // One byte of annotations.
+        dataBuilder.write(0x85); // Annotation. Conveniently use SID 5 ("version"), which is in the system symbol table.
+        dataBuilder.write(0xCE); // S-exp with length subfield.
+        IonBinary.writeVarUInt(dataBuilder, raw.length);
+        dataBuilder.write(raw);
+        byte[] data = dataBuilder.toByteArray();
+
+        final int totalNumberOfBatches = (Integer.MAX_VALUE / data.length) + 1000000;
+        ByteArrayOutputStream header = new ByteArrayOutputStream();
+        header.write(BINARY_VERSION_MARKER_1_0);
+        long containerLength = (long) data.length * totalNumberOfBatches;
+        header.write(0xEE); // Annotation wrapper with length subfield.
+        IonBinary.writeVarUInt(header, containerLength + 3 + IonBinary.lenVarUInt(containerLength));
+        header.write(0x81); // One byte of annotations.
+        header.write(0x84); // Conveniently use SID 4 ("name"), which is in the system symbol table.
+        header.write(0xBE); // List with length subfield.
+        IonBinary.writeVarUInt(header, containerLength); // Length
+        InputStream inputStream = new SequenceInputStream(
+            new ByteArrayInputStream(header.toByteArray()),
+            new RepeatInputStream(data, totalNumberOfBatches - 1) // This will provide the data 'totalNumberOfBatches' times
+        );
+        IonReader reader = IonReaderBuilder.standard().build(inputStream);
+        assertEquals(IonType.LIST, reader.next());
+        String[] annotations = reader.getTypeAnnotations();
+        assertEquals(1, annotations.length);
+        assertEquals("name", annotations[0]);
+        assertNull(reader.next());
+    }
+
+    // Note: the objective of the following tests is not to assert that large scalars *should* fail, but rather that
+    // when they *do* fail due to limitations of the current implementation, they fail by throwing an IonException
+    // and not something unexpected and ugly.
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    private void cleanlyFailsOnLargeScalar(IonReaderBuilder readerBuilder) throws Exception {
+        byte[] data = "foobarbaz".getBytes("UTF-8");
+        final int totalNumberOfBatches = (Integer.MAX_VALUE / data.length) + 123;
```

Comment:
What's the significance of `123`?

Reply:
It's an arbitrary number that causes the size of the value to exceed Integer.MAX_VALUE. I can add a comment to that effect.